STM32 USB update step 1

jeromecoutant opened the pull request:
### Description

Hi

Goal is to make USB Device integration easy for each STM32 targets.

Step 1 (this PR):
- USB pins are declared for all current supported ST boards
even if USB_DEVICE feature is not supported yet
even is the board doesn't have any connector
- So, no more need to manually hardcode pins value for each target 
This fixes #10806 
Only addition of USB_DEVICE in the local mbed_app.json file is necessary
- removes compilation warning
This fixes #10822
- this should work for every targets supporting OTG and with full speed connector
**So targets, which was supporting USB, still supports USB (no removed target, no target added)**
- Old and not supported MBED2 USB lib is removed from targets.json

Next step (other PR coming soon):
- STM32WB, STM32H7, STM32L0
- USB High speed 

@bulislaw @MarceloSalazar @LMESTM 

### Pull request type

    [ ] Fix
    [x] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

### Release Notes

jeromecoutant committed:
STM32 gen PeriphPin script update for USB

jeromecoutant committed:
STM32 PeripheralPins.h update with USB

jeromecoutant committed:
STM32F0 USB pins addition

jeromecoutant committed:
STM32F1 USB pins addition

jeromecoutant committed:
STM32F2 USB pins addition

jeromecoutant committed:
STM32F3 USB pins addition

jeromecoutant committed:
STM32F4 USB pins addition

jeromecoutant committed:
STM32F7 USB pins addition

jeromecoutant committed:
STM32H7 USB pins addition

jeromecoutant committed:
STM32L0 USB pins addition

jeromecoutant committed:
STM32L1 USB pins addition

jeromecoutant committed:
STM32L4 USB pins addition

jeromecoutant committed:
STM32WB USB pins addition

jeromecoutant committed:
STM32 clock configuration depending on USB

jeromecoutant committed:
STM32 targets.json update for USB

Removed unsupported MBED2 features:
- USB_STM_HAL
- USBHOST_OTHER

mark-edgeworth commented on tools/targets/STM32_gen_PeripheralPins.py:
```diff
@@ -1,6 +1,6 @@
 """
 * mbed Microcontroller Library
-* Copyright (c) 2006-2018 ARM Limited
+* Copyright (c) 2019 ARM Limited
```
This should be 2006-2019, unless you are sure that none of this was developed in the years inbetween.

mark-edgeworth commented on tools/targets/STM32_gen_PeripheralPins.py:
```diff
@@ -332,7 +334,12 @@ def store_qspi(pin, name, signal):
+    if signal.startswith("USB_OTG_FS"):
+        usb_otgfs_list.append([pin, name, signal])
+    if signal.startswith("USB_OTG_HS"):
+        usb_otghs_list.append([pin, name, signal])
```
This could be incorrect as it changes the behaviour for all signals that contain 'OTG' that do not match the special case. Whilst there may be none of these today, this will fail to add the pin in future if the signal contains 'OTG'. Suggest an if..elif..elif...else construct could be better here.
If this is not the intent then there should be a big comment explaining why. Ideally there should be a comment to explain what this new code is trying to do in any case.

mark-edgeworth commented on tools/targets/STM32_gen_PeripheralPins.py:
```diff
@@ -873,8 +888,80 @@ def print_qspi(l):
+    use_hs_in_fs = False
+    nb_loop = 1
+    inst = "USB_FS"
+    if lst == usb_otgfs_list:
```
This is a list element comparison - use 'is' instead. However, the pattern of accessing a global variable from inside the function in this way makes the code harder to understand, harder to maintain and harder to test.

mark-edgeworth commented on tools/targets/STM32_gen_PeripheralPins.py:
```diff
@@ -873,8 +888,80 @@ def print_qspi(l):
+    inst = "USB_FS"
+    if lst == usb_otgfs_list:
+        inst = "USB_FS"
+    elif lst == usb_otghs_list:
```
same here.

mark-edgeworth commented on tools/targets/STM32_gen_PeripheralPins.py:
```diff
@@ -873,8 +888,80 @@ def print_qspi(l):
+            s1 += "\n"
+            out_c_file.write(s1)
+    if lst:
+        if lst == usb_otghs_list:
```
same here

jeromecoutant committed:
STM32 USB redesign step 1

No more need to explicitly configure each targets.
Pins are now defined in the PeripheralPin.c file which is build by a script.
